eventdiscovery-sdk/src/main/java/com/schedjoules/eventdiscovery/framework/widgets/OptionalTextedView.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright 2017 SchedJoules
+ *
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.schedjoules.eventdiscovery.framework.widgets;
+
+import android.view.View;
+import android.widget.TextView;
+
+import com.schedjoules.eventdiscovery.framework.utils.smartview.SmartView;
+
+import org.dmfs.optional.Optional;
+
+
+/**
+ * Represents a View that has an updatable text, and which should only be shown if the text value is available.
+ *
+ * @author Gabor Keszthelyi
+ */
+public final class OptionalTextedView implements SmartView<Optional<CharSequence>>
+{
+    private final OptionalView mRoot;
+    private final TextView mTextView;
+
+
+    public OptionalTextedView(View root, TextView textView)
```

Comment:
why does it need both, the View and the Root? Could you pass the view resource id instead of the view itself?